docs/source/advanced/mixed_precision.rst

```diff
@@ -5,49 +5,158 @@
+complete accuracy and employing a 32-bit floating-point for those steps only, while using a 16-bit floating-point for the rest. When compared to complete precision training, mixed precision training
+delivers all of these benefits while ensuring that no task-specific accuracy is lost. [`2 <https://docs.nvidia.com/deeplearning/performance/mixed-precision-training/index.html>`_].
 
 .. note::
 
-    In some cases it is important to remain in FP32 for numerical stability, so keep this in mind when using mixed precision.
+    In some cases, it is essential to remain in FP32 for numerical stability, so keep this in mind when using mixed precision.
+    For example, when running scatter operations during the forward (such as torchpoint3d), computation must remain in FP32.
+
+.. warning::
+
+    Do not cast anything to other dtypes manually using ``torch.autocast`` or ``tensor.half()`` when using native precision because
+    this can bring instability.
+
+    .. code-block:: python
+
+        class LitModel(LightningModule):
+            def training_step(self, batch, batch_idx):
+                outs = self(batch)
+
+                a_float32 = torch.rand((8, 8), device=self.device, dtype=self.dtype)
+                b_float32 = torch.rand((8, 4), device=self.device, dtype=self.dtype)
+
+                # casting to float16 manually
+                with torch.autocast(device_type=self.device.type):
+                    c_float16 = torch.mm(a_float32, b_float32)
+                    target = self.layer(c_float16.flatten()[None])
+
+                # here outs is of type float32 and target is of type float16
+                loss = torch.mm(target @ outs).float()
+                return loss
+
+
+        trainer = Trainer(gpus=1, precision=32)
 
-    For example when running scatter operations during the forward (such as torchpoint3d) computation must remain in FP32.
 
 FP16 Mixed Precision
---------------------
+====================
 
-In most cases, mixed precision uses FP16. Supported torch operations are automatically run in FP16, saving memory and improving throughput on GPU and TPU accelerators.
+In most cases, mixed precision uses FP16. Supported `PyTorch operations <https://pytorch.org/docs/stable/amp.html#op-specific-behavior>`__ automatically run in FP16, saving memory and improving throughput on the supported accelerators.
 
-Since computation happens in FP16, there is a chance of numerical instability during training. This is handled internally by a dynamic grad scaler which skips steps that are invalid, and adjusts the scaler to ensure subsequent steps fall within a finite range. For more information `see the autocast docs <https://pytorch.org/docs/stable/amp.html#gradient-scaling>`__.
 
 .. note::
 
-    When using TPUs, setting ``precision=16`` will enable bfloat16 which is the only supported precision type on TPUs.
+    When using TPUs, setting ``precision=16`` will enable bfloat16, the only supported mixed precision type on TPUs.
```

Comment:
```suggestion
    When using TPUs, setting ``precision=16`` will enable bfloat16, the only supported half precision type on TPUs.
```

I think that for TPUs bfloat16 is no mixed precision but used everywhere if enabled cc @kaushikb11 to confirm

Reply:
confirmed!